src/prommis/nanofiltration/tests/test_nf_brine.py

```diff
@@ -0,0 +1,47 @@
+#################################################################################
+# adapted from test_nf.py
+#################################################################################
+
+"""
+Tests the nanofiltration flowsheet
+"""
+
+from pyomo.environ import value
+
+import pytest
+
+pytest.importorskip("watertap", reason="WaterTAP dependency not available")
+# pylint: disable-next=wrong-import-position
+from prommis.nanofiltration.nf_brine import main
+
+
+@pytest.mark.component
+def test_main():
+    """
+    Tests the execution of the main function in nf_brine.py
+    """
+    m = main()
```

Comment:
It might not be possible for this model (as it is rather complicated and comes from WaterTAP), but it would be nice to look at adding diagnostics checks in here too (`dt.assert_no_strucutral_warnings()` and `dt.assert_no_numerical_warnings()`).

This should probably wait until a later date however, as it will likely require some work.